kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
In my opinion, both this and `CommitCoordinatorBuilder` should be moved to the kernel-defaults implementation PR so that this PR can mostly focus on the main interface additions.